### DISCUSSION TITLE
Opposite of collides

### DISCUSSION THREAD

**Discussion by pika11** [replit/kaboom#396] [Q&A] [answered]
Is there a similar event like `collides` but for when a collision ends? Maybe the `collide` callback could return a function for when the collision ends? 

I’m guessing the way to do this would be on `action` and check if there are any collisions? or am I missing something? 

thanks in advance! kaboom is really cool! 🚀

**Comment by slmjkdbtl** [comment, marked as answer]
You mean an event when an object stops colliding with another? 

Yeah the way to do it now would be in an `action` (which is deprecated in favor of `onUpdate` now) and check for `isColliding()`\

But I think it's possible to support that built-in, since kaboom are also already doing some [internal bookkeeping](https://github.com/replit/kaboom/blob/daf10ee6e65b83e7dae92e58f6e259957282fb6c/src/kaboom.ts#L1457-L1462) about if 2 collided objects are still colliding.

**Reply by pika11** [reply]
yeah exactly. Thanks! 

would be cool to have it just to avoid the extra “manual” checks.

**Reply by slmjkdbtl** [reply]
Yeah I'd like that too. Do you happen to have name suggestions? I'm thinking something like `onCollideExit` or `onCollideEnd` but not sure yet

**Reply by pika11** [reply]
tough call! naming things is hard 😄. I would lean towards `onCollideEnd`.

great job once again on kaboom, docs are 💯

**Comment by oaklandgit** [comment]
@pika11 -- In case it's helpful, I had to solve this problem today so I tried @slmjkdbtl's suggestion. This works:

(I want to show elements when the "player" is standing in front of a "computer", but remove them after he's moved away.)

```js
export const computer = () => [
    k.sprite("computer"),
    k.area(),
    {
        add() {
            this.onCollide("player", (p) => {
                console.log("player hovering")
                this.objColliding = p
            })

            this.onUpdate(() => {
                if (!this.objColliding) return
                if(!this.isColliding(this.objColliding)) {
                    console.log("player leaving")
                    this.objColliding = null
                }
            })
        }
    }
```
